### ISSUE TITLE
not working

### ISSUE THREAD

**Issue by kryper23** [coinables/xpub-donation#1]
hello the new update is not working

**Comment by coinables** [comment]
Please be specific. What is not working?

There must be at least one existing transaction on the provided xpub key chain for the new API to populate the next address. This is a limitation of the API.

**Comment by kryper23** [comment]
i try your script but the next address is not showing. i dont know why..

**Comment by coinables** [comment]
Ok, do me a favor. What happens when you paste your xpub in the search bar at blockchair.com? Does it recognize any existing transactions on that xpub?

**Comment by kryper23** [comment]
Yes. I try to search my xpub key to blockchair and i see my transaction.
Then i paste my xpub key to your new update script and install to my
web,but no address appears. Do you have a video tutorial?

**Comment by coinables** [comment]
I'm unable to reproduce your error. I assume  you are running xampp or wampp or other apache server?

**Comment by kryper23** [comment]
I use hostinger

**Comment by EAWF** [comment]
Is it possible that an api key is needed?

**Comment by kryper23** [comment]
Yes.coz the free api have a limit

**Comment by GPUheroes** [comment]
Is there any chance you could create a version of Xpub-donation that displays a new receiving address whenever a Donation button is pressed so there's no need to rely on blockchair.com or other external service? There will likely be times people will press the Donation button without ever sending anything to the receiving address that's presented. But a website owner who receives only occasional donations is unlikely to ever run out of receiving addresses despite some addresses remaining unused. Perhaps the code could limit the number of new addresses obtained by each visitor IP number to prevent a visitor from accidentally or intentionally tying up all the receiving addresses.

If this is something you don't want to tackle in the near future, are you aware of any alternative existing code that displays a new receiving address without relying on external services and without being tied to a specific host such as WordPress?

Thanks
